src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1503 @@ const ActorInit En_Elf_InitVars = {
+            case 7:
+                player->stateFlags2 &= ~0x100000;
+                break;
+            default:
+                player->stateFlags2 |= 0x100000;
+                break;
+        }
+    }
+
+    if (temp != this->unk_2A8) {
+        func_80A01C38(this, temp);
+
+        if (temp == 11) {
+            this->unk_2B8 = Math_Vec3f_DistXZ(&player->bodyPartsPos[8], &this->actor.posRot.pos);
+            this->unk_2AC = Math_Vec3f_Yaw(&this->actor.posRot.pos, &player->bodyPartsPos[8]);
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A0461C.s")
+#endif
+
+void EnElf_SpawnSparkles(EnElf* this, GlobalContext* globalCtx, s32 sparkleLife) {
+    static Vec3f sparkleVelocity = { 0.0f, -0.05f, 0.0f };
+    static Vec3f sparkleAccel = { 0.0f, -0.025f, 0.0f };
+    s32 pad;
+    Vec3f sparklePos;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+
+    sparklePos.x = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.x;
+    sparklePos.y = (Math_Rand_ZeroOne() * 6.0f) + this->actor.posRot.pos.y;
+    sparklePos.z = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.z;
+
+    primColor.r = this->innerColor.r;
+    primColor.g = this->innerColor.g;
+    primColor.b = this->innerColor.b;
+
+    envColor.r = this->outerColor.r;
+    envColor.g = this->outerColor.g;
+    envColor.b = this->outerColor.b;
+
+    EffectSsKiraKira_SpawnDispersed(globalCtx, &sparklePos, &sparkleVelocity, &sparkleAccel, &primColor, &envColor,
+                                    1000, sparkleLife);
+}
+
+void func_80A04D90(EnElf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 bgId;
+
+    this->actor.groundY = func_8003CA0C(globalCtx, &globalCtx->colCtx, &this->actor.floorPoly, &bgId, &this->actor,
+                                        &this->actor.posRot.pos);
+    this->actor.shape.unk_14 = 0x32;
+}
+
+// move to talk to player
+void func_80A04DE4(EnElf* this, GlobalContext* globalCtx) {
+    Vec3f posRot2Copy;
+    Player* player = PLAYER;
+    Vec3f naviRefPos;
+
+    if (this->fairyFlags & 0x10) {
+        naviRefPos = globalCtx->actorCtx.targetCtx.naviRefPos;
+
+        if ((player->unk_664 == NULL) || (&player->actor == player->unk_664) || (&this->actor == player->unk_664)) {
+            naviRefPos.x = player->bodyPartsPos[7].x + (Math_Sins(player->actor.shape.rot.y) * 20.0f);
+            naviRefPos.y = player->bodyPartsPos[7].y + 5.0f;
+            naviRefPos.z = player->bodyPartsPos[7].z + (Math_Coss(player->actor.shape.rot.y) * 20.0f);
+        }
+
+        this->actor.posRot2.pos = naviRefPos;
+        this->fairyFlags &= ~0x10;
+    }
+
+    func_80A03AB0(this, globalCtx);
+    posRot2Copy = this->actor.posRot2.pos;
+
+    func_80A03148(this, &posRot2Copy, 0, 20.0f, 0.2f);
+
+    if (this->actor.speedXZ >= 5.0f) {
+        EnElf_SpawnSparkles(this, globalCtx, 16);
+    }
+
+    Math_SmoothScaleMaxMinF(&this->actor.scale.x, 0.008f, 0.3f, 0.00080000004f, 0.000080000005f);
```

Comment:
```suggestion
    Math_SmoothScaleMaxMinF(&this->actor.scale.x, 0.008f, 0.3f, 0.0008f, 0.00008f);
```

Reply:
these dont match for some reason. it should also be `(1.0f / 1250.0f)` and `(1.0f / 12500.0f)` respectively but they are off by 1 in hex

Reply:
In that case my guess is they are `1.1f /` instead of `1.0f /`

Reply:
More likely they are `1.0 / 1250` and `1.0 / 12500`. These off-by-ones are usually converted doubles.

Reply:
neither of these suggestions are working lol, wtf
I cant get it to go without the really long decimal

Reply:
yea im not sure what to do with these floats, we may be able to clean them up at a later time